### PR TITLE
restart watch of endpoint when node goes away

### PR DESCRIPTION
**Pull request by jplouis**

### CODE REVIEW

File: container/endpoint.go [hunk 1]
```diff
@@ -361,6 +361,17 @@ func (c *Controller) watchRemotePorts() {
 		return
 	}
 
+	//translate closing call to endpoint cancel
+	cancelEndpointWatch := make(chan bool)
+	go func() {
+		select {
+		case <-c.closing:
+			glog.Infof("Closing endpoint watchers")
+			close(endpointsWatchCanceller)
```
Comment (smousa) [inline container/endpoint.go]: Is this a race condition?  You are writing to this channel below...

File: container/endpoint.go [hunk 2]
```diff
@@ -380,17 +391,31 @@ func (c *Controller) watchRemotePorts() {
 			}
 			if !missingWatchers {
 				glog.V(2).Infof("all imports are being watched - cancelling watcher on /endpoints")
-				endpointsWatchCanceller <- true
-				return
+				select{
+				case endpointsWatchCanceller <- true:
```
Comment (smousa) [inline container/endpoint.go]: You write to this channel here

File: container/endpoint.go (hunk 1 above)
Comment (jplouis) [inline container/endpoint.go]: the default should take care of it.

File: container/endpoint.go [hunk 3]
```diff
@@ -380,17 +391,31 @@ func (c *Controller) watchRemotePorts() {
...
+					return
+				default:
```
Comment (jplouis) [inline container/endpoint.go]: Default

File: container/controller.go [hunk 4]
```diff
@@ -630,6 +630,7 @@ func (c *Controller) Run() (err error) {
 		return nil
 	}
 	defer client.Close()
+	c.Close()
```
Comment (jplouis) [inline container/controller.go]: This was added to let things know it is shutting down.

**Pull request closed without merging**